sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/options/SetKeyVaultRoleDefinitionOptions.java

```diff
@@ -0,0 +1,143 @@
+        return roleScope;
+    }
+
+    /**
+     * Get the name of the {@link KeyVaultRoleDefinition} to create or update.
+     *
+     * @return The name of the {@link KeyVaultRoleDefinition}.
+     */
+    public String getRoleDefinitionName() {
+        return roleDefinitionName;
+    }
+
+    /**
+     * Get the role name.
+     *
+     * @return The role name.
+     */
+    public String getRoleName() {
+        return roleName;
+    }
+
+    /**
+     * Set the role name.
+     *
+     * @param roleName The role name to set.
+     *
+     * @return The updated {@link SetKeyVaultRoleDefinitionOptions} object.
+     */
+    public SetKeyVaultRoleDefinitionOptions setRoleName(String roleName) {
+        this.roleName = roleName;
+
+        return this;
+    }
+
+    /**
+     * Get the role description.
+     *
+     * @return The role description.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Set the role description.
+     *
+     * @param description The role description to set.
+     *
+     * @return The updated {@link SetKeyVaultRoleDefinitionOptions} object.
+     */
+    public SetKeyVaultRoleDefinitionOptions setDescription(String description) {
+        this.description = description;
+
+        return this;
+    }
+
+    /**
+     * Get the {@link KeyVaultRoleDefinition role definition}'s {@link KeyVaultPermission permissions}.
+     *
+     * @return The {@link KeyVaultRoleDefinition role definition}'s {@link KeyVaultPermission permissions}.
+     */
+    public List<KeyVaultPermission> getPermissions() {
+        return permissions;
+    }
+
+    /**
+     * Set the {@link KeyVaultRoleDefinition role definition}'s {@link KeyVaultPermission permissions}.
+     *
+     * @param permissions The {@link KeyVaultRoleDefinition role definition}'s {@link KeyVaultPermission permissions}
+     * to set.
+     *
+     * @return The updated {@link SetKeyVaultRoleDefinitionOptions} object.
+     */
+    public SetKeyVaultRoleDefinitionOptions setPermissions(List<KeyVaultPermission> permissions) {
+        this.permissions = permissions;
+
+        return this;
+    }
+
+    /**
+     * Get the {@link KeyVaultRoleDefinition role definition}'s assignable scopes.
+     *
+     * @return The {@link KeyVaultRoleDefinition role definition}'s assignable scopes.
+     */
+    public List<KeyVaultRoleScope> getAssignableScopes() {
+        return assignableScopes;
+    }
+
+    /**
+     * Set the {@link KeyVaultRoleDefinition role definition}'s assignable scopes.
+     *
+     * @param assignableScopes The {@link KeyVaultRoleDefinition role definition}'s assignable scopes to set.
+     *
+     * @return The updated {@link SetKeyVaultRoleDefinitionOptions} object.
+     */
+    public SetKeyVaultRoleDefinitionOptions setAssignableScopes(List<KeyVaultRoleScope> assignableScopes) {
+        this.assignableScopes = assignableScopes;
+
+        return this;
+    }
+}
```

Comment:
Should also have a `getRoleType` - not settable, though. As mentioned in API unification, you can set this via the method. It's BUILT_IN_ROLE unless it was actually created by the method, in which case it's CUSTOM_ROLE. Should also be returned from the service as such since it's in swagger.